tests/e2e/database/ref/on.e2e.js

```diff
@@ -0,0 +1,45 @@
+const { CONTENTS, setDatabaseContents } = TestHelpers.database;
+const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+describe('database()', () => {
+  before(() => setDatabaseContents());
+  describe('ref().on()', () => {
+    it('throws without second argument', () => {
+      const ref = firebase.database().ref('tests/types/number');
+      try {
+        const returnValue = ref.on('value');
```

Comment:
You could use `should.throw` to avoid needing a try catch, e.g. `(() => ref.on('value')).should.throw('The error message that it should throw');`